query/src/main/java/io/keen/client/java/SavedQueriesImpl.java

```diff
@@ -0,0 +1,396 @@
+                                  null /* miscProperties */);
+
+
+        return queryClient.getMapResponse(newSavedQueryRequest);
+    }
+
+    @Override
+    public Map<String, Object> createCachedQuery(String queryName,
+                                                 KeenQueryRequest query,
+                                                 int refreshRate) throws IOException {
+        PersistentAnalysis newCachedQueryRequest =
+                new SavedQueryPut(queryName,
+                                  null /* displayName */,
+                                  query,
+                                  refreshRate,
+                                  null /* miscProperties */);
+
+        return queryClient.getMapResponse(newCachedQueryRequest);
+    }
+
+    @Override
+    public Map<String, Object> createCachedQuery(String queryName,
+                                                 KeenQueryRequest query,
+                                                 String displayName,
+                                                 int refreshRate) throws IOException {
+        PersistentAnalysis newCachedQueryRequest =
+                new SavedQueryPut(queryName,
+                                  displayName,
+                                  query,
+                                  refreshRate,
+                                  null /* miscProperties */);
+
+        return queryClient.getMapResponse(newCachedQueryRequest);
+    }
+
+    @Override
+    public Map<String, Object> getDefinition(String queryName) throws IOException {
+        PersistentAnalysis getDefRequest = new SavedQueryRequest(HttpMethods.GET,
+                                                                 true /* needsMasterKey */,
+                                                                 queryName);
+        if (null == queryName) {
+            // Note that the PersistentAnalysis class will further validate queryName.
+            throw new IllegalArgumentException("A query name is required.");
+        }
+
+        // When retrieving a single query definition, we expect a single JSON Object and just
+        // hand it back instead of wrapping in QueryResult.
+        return queryClient.getMapResponse(getDefRequest);
+    }
+
+    @Override
+    public List<Map<String, Object>> getAllDefinitions() throws IOException {
+        PersistentAnalysis getAllDefsRequest = new SavedQueryRequest(HttpMethods.GET,
+                                                                     true /* needsMasterKey */,
+                                                                     null /* queryName */);
+        List<Object> response = queryClient.getListResponse(getAllDefsRequest);
+
+        // We expect a structure such that each entry in the list was a JSON Object representing a
+        // query definition, and no entry should be a JSON Value.
+        for (Object defObj : response) {
+            if (!(defObj instanceof LinkedHashMap)) {
```

Comment:
I was intentionally more specific only because I happen to know empirically that `LinkedHashMap` is what that lib should produce. But, you're right in that we could be more lenient and future-proof if we accepted the most abstract interface that provides the necessary functionality, so I'll change it. Thanks!